Services/UserService/Commands/EditUserCommand.cs

```diff
@@ -11,29 +16,70 @@ public class EditUserCommand : IEditUserCommand
     {
         private readonly IValidator<UserInfoRequest> userInfoValidator;
         private readonly IValidator<PasswordChangeRequest> passwordChangeValidator;
-        
+        private readonly IBus busControl;
+
         public EditUserCommand
             ([FromServices] IValidator<UserInfoRequest> userInfoValidator, 
-            [FromServices] IValidator<PasswordChangeRequest> passwordChangeValidator)
+            [FromServices] IValidator<PasswordChangeRequest> passwordChangeValidator,
+            [FromServices] IBus busControl)
         {
+            this.busControl = busControl;
             this.userInfoValidator = userInfoValidator;
             this.passwordChangeValidator = passwordChangeValidator;
         }
         
-        public Task<bool> Execute(EditUserRequest request)
+        public async Task<bool> Execute(EditUserRequest request)
         {
             var passwordRequest = request.PasswordRequest;
             var userInfo = request.UserInfo;
-            
+            PasswordHashChangeRequest passwordHashChangeRequest = null;
+
             if (!(string.IsNullOrEmpty(passwordRequest.OldPassword)
                     || string.IsNullOrEmpty(passwordRequest.NewPassword)))
             {
                 passwordChangeValidator.ValidateAndThrow(passwordRequest);
+                passwordHashChangeRequest = new PasswordHashChangeRequest
+                {
+                     OldPasswordHash = ShaHash.GetPasswordHash(passwordRequest.NewPassword),
+                     NewPasswordHash = ShaHash.GetPasswordHash(passwordRequest.OldPassword)
+                };
             }
+           
             userInfoValidator.ValidateAndThrow(userInfo);
 
-            // TODO Realise edit user logic
-            throw new NotImplementedException();
+            var user = new User
```

Comment:
Why do wee need this variable? Can't we create new `User `while creating `InternalEditUserInfoRequest`?

Reply:
Because I wanted EditUser to be one style with CreateUser